app/src/main/java/kantwonskids/donationtrackerg14b/controller/InventoryActivity.java

```diff
@@ -16,31 +16,41 @@
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
 import android.widget.TextView;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import kantwonskids.donationtrackerg14b.R;
-import kantwonskids.donationtrackerg14b.model.Donation;
-import kantwonskids.donationtrackerg14b.model.Model;
+import kantwonskids.donationtrackerg14b.model.*;
 
 /**
  * an Activity to represent the inventory (list of donation items)
  * at a given location
  */
 public class InventoryActivity extends AppCompatActivity {
 
+    private User user = Model.getInstance().getCurrentUser();
+    private UserRole role = user.getUserRole();
+    private Location location = user.getLocation();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_inventory);
 
         FloatingActionButton addItemButton = findViewById(R.id.addItem);
-        addItemButton.setOnClickListener((view) -> {
-            Intent intent_addToInventory = new Intent(this, NewItemActivity.class);
-            startActivity(intent_addToInventory);
-        });
+        if ((role == UserRole.LOCATION_EMPLOYEE
```

Comment:
There are methods within the UserRole enum to do this (instead of ==)

Reply:
I just checked and there is no method for canAddOrRemoveItems

Reply:
ah ok so canUpdateInfo() is that. you might wanna rename it if you actually use it because it's not a very good name lol.

also, I could've sworn there was a method canUpdateInfoAt(Location loc) for location employees, but i guess not. good design would be to have that method defined in the User class. that might be useful to have.